src/Network/NWProtocolTcpOptions.cs

```diff
@@ -0,0 +1,113 @@
+//
+// NWProtocolTcp: Bindings the Netowrk nw_protocol_options API focus on Tcp options.
+//
+// Authors:
+//   Manuel de la Pena <mandel@microsoft.com>
+//
+// Copyrigh 2019 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+using Security;
+using OS_nw_protocol_definition=System.IntPtr;
+using IntPtr=System.IntPtr;
+
+namespace Network {
+
+	[TV (12,0), Mac (10,14), iOS (12,0), Watch (6,0)]
+	public class NWProtocolTcpOptions : NWProtocolOptions {
+		
+		internal NWProtocolTcpOptions (IntPtr handle, bool owns) : base (handle, owns) {}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern IntPtr nw_tcp_create_options ();
+		
+		public NWProtocolTcpOptions () : this (nw_tcp_create_options (), owns: true) {}
+
+		[DllImport (Constants.NetworkLibrary)]
+		extern static void nw_tcp_options_set_no_delay (IntPtr handle, [MarshalAs (UnmanagedType.U1)] bool noDelay);
+
+		public void SetNoDelay (bool noDelay) => nw_tcp_options_set_no_delay (GetCheckedHandle (), noDelay);
+
+		[DllImport (Constants.NetworkLibrary)]
+		extern static void nw_tcp_options_set_no_push (IntPtr handle, [MarshalAs (UnmanagedType.U1)] bool noPush);
```

Comment:
you're duplicating the `p/invoke` declarations (this one and others)
since it's internal there's no need to dupe them, just call the old one
or make the deprecated code call those new ones